lib/instance.js

```diff
@@ -640,16 +641,24 @@ Instance.prototype.save = function(options) {
         self.dataValues[createdAtAttr] = values[self.Model.rawAttributes[createdAtAttr].field || createdAtAttr] = self.Model.$getDefaultTimestamp(createdAtAttr) || now;
       }
 
+      var tableName;
+      if (self.$options.$schema !== self.Model.$schema ||
+        self.$options.$schemaDelimiter !== self.Model.$schemaDelimiter) {
+        tableName = self.Model.schema(self.$options.$schema, self.$options.$schemaDelimiter).getTableName(options);
+      } else {
```

Comment:
This is not odd at all if I am trying to sue the same model across schemas. The model is attached to the Instance prototype when init is called on the model: https://github.com/sequelize/sequelize/blob/master/lib/model.js#L724-L725. If I then call .schema on this model it will not change $schema on the model attached to the prototype.